crates/bevy_render/src/camera/projection.rs

```diff
@@ -134,54 +134,71 @@ impl Default for PerspectiveProjection {
+    /// The argument is the number of pixels that equals one world unit.
+    WindowSize(f32),
+    /// Use the minimal possible projection size while keeping the aspect ratio.
     /// Arguments are in world units.
     Auto { min_width: f32, min_height: f32 },
-    /// Keep vertical axis constant; resize horizontal with aspect ratio.
-    /// The argument is the desired height of the viewport in world units.
+    /// Keep the projection's height constant; width will be adjusted to match aspect ratio.
+    /// The argument is the desired height of the projection in world units.
     FixedVertical(f32),
-    /// Keep horizontal axis constant; resize vertical with aspect ratio.
-    /// The argument is the desired width of the viewport in world units.
+    /// Keep the projection's width constant; height will be adjusted to match aspect ratio.
+    /// The argument is the desired width of the projection in world units.
     FixedHorizontal(f32),
 }
 
+/// Project a 3D space onto a 2D surface using parallel lines, i.e., unlike [`PerspectiveProjection`],
+/// the size at which objects appear remain the same regardless of depth.
+///
+/// The volume contained in the projection is called the *view frustum*. Since the viewport is rectangular,
+/// the view frustum is in the shape of a rectangular prism.
+///
+/// Note that the cross sectional area of the view frustum and the apparent size of objects are inversely proportional.
 #[derive(Component, Debug, Clone, Reflect, FromReflect)]
 #[reflect(Component, Default)]
 pub struct OrthographicProjection {
-    pub left: f32,
-    pub right: f32,
-    pub bottom: f32,
-    pub top: f32,
+    /// The distance of the near clipping plane in world units.
+    ///
+    /// Objects closer than this will not be rendered.
     pub near: f32,
+    /// The distance of the far clipping plane in world units.
+    ///
+    /// Objects further than this will not be rendered.
     pub far: f32,
-    pub window_origin: WindowOrigin,
+    /// Specifies the origin of the viewport as a fraction of its width and height (from the bottom left corner).
+    /// Consequently, this is the point from where scaling caused by viewport resizing will occur.
+    ///
+    /// For example, if `viewport_origin` is set to (0.2, 0.6) and the size of the viewport is (10, 10),
+    /// the location of the camera will determine where the point (2, 6) on the viewport is;
+    /// if the projection needs to triple in width, it will expand 4 units to the left and
+    /// 16 units to the right (20% and 80% of the total scaling, respectively).
+    pub viewport_origin: (f32, f32),
+    /// How the projection will scale when the viewport is resized.
     pub scaling_mode: ScalingMode,
+    /// Scales the projection, in world units.
+    ///
+    /// As scale increases, the apparent size of objects decreases, and vice versa.
     pub scale: f32,
+    pub left: f32,
+    pub right: f32,
+    pub bottom: f32,
+    pub top: f32,
```

Comment:
I've left these public for now. Would it be better to make them private and change the way `OrthographicProjection` is initialized? `PerspectiveProjection` should probably also be changed in that case, but that might be for the better since it also has what's essentially a read-only public field.

Reply:
I'm in favor.